subworkflows/nf-core/bam_dedup_stats_samtools_umitools/main.nf

```diff
@@ -0,0 +1,56 @@
+//
+// UMI-tools dedup, index BAM file and run samtools stats, flagstat and idxstats
+//
+
+include { UMITOOLS_DEDUP     } from '../../../modules/nf-core/umitools/dedup/main'
+include { SAMTOOLS_INDEX     } from '../../../modules/nf-core/samtools/index/main'
+include { BAM_STATS_SAMTOOLS } from '../bam_stats_samtools/main'
+
+workflow BAM_DEDUP_STATS_SAMTOOLS_UMITOOLS {
+    take:
+    bam_bai         // channel: [ val(meta), [ bam ], [ bai/csi ] ]
+    get_dedup_stats // boolean: true/false
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    //
+    // UMI-tools dedup
+    //
+    UMITOOLS_DEDUP ( bam_bai, get_dedup_stats )
+    ch_versions = ch_versions.mix(UMITOOLS_DEDUP.out.versions.first())
+
+    //
+    // Index BAM file and run samtools stats, flagstat and idxstats
+    //
+    SAMTOOLS_INDEX ( UMITOOLS_DEDUP.out.bam )
```

Comment:
I think we can assume that. This module origins from https://github.com/nf-core/rnaseq/blob/master/subworkflows/nf-core/dedup_umi_umitools.nf - It works nicely there. 
@drpatelh Opinion?

Reply:
By default it will sort the BAM file as output:
https://umi-tools.readthedocs.io/en/latest/reference/dedup.html#no-sort-output

We could maybe update it to deal with unsorted output when the use case arises and is probably out of the scope of this PR. But for this we will probably need to parameterise `--no-sort-output` and add it as an input to the upstream module. Then have an input in the subworklow that determines whether or not to run the indexing.